src/libImaging/Quant.c

```diff
@@ -789,6 +789,44 @@ resort_distance_tables(
     return 1;
 }
 
+/* build_distance_tables() below used to call qsort() from stdlib.h.
+ * qsort() is not guaranteed stable, so different implementations can
+ * give different results, causing images created on Linux and Windows
+ * to differ. Shell sort is also not stable, but having the implementation
+ * included here will give consistent results across platforms. -rdg- */
+
+/*  ssort()  --  Fast, small, qsort()-compatible Shell sort */
+/*  By Raymond D. Gardner. Released here under PIL license. */
+static void
+ssort(void  *base, uint32_t nmemb, uint32_t size,
+        int (*compar)(const void *, const void *)) {
+    uint32_t sz_nmemb, gap, sz_gap, i, j, k;
+    char *a, *b, tmp;
+
+    sz_nmemb = size * nmemb;
+    for (gap = 0; ++gap < nmemb;)
+        gap *= 3;
```

Comment:
```suggestion
    for (gap = 0; ++gap < nmemb;) {
        gap *= 3;
    }
```
As in #4618

Reply:
Thanks for looking at my code. I usually use mostly K&R style. I noticed how your ClangFormat handled the function opening braces and changed that (looks weird to me) but missed that you fully brace single-statement `for/if` bodies. Fixed. I also extracted and re-tested the `ssort()` function.
(Now you might consider changing this: `    for (;;) switch (state->state) {` line 172 in `GifEncode.c`. This is a huge one-statement `for` body.)
You tagged this "Needs Tests". What sort of test would you propose? Not sure how one could test this. The sort itself is very well tested.

Reply:
If, at some point in the future, for whatever reason, we removed your change, all of our tests would still pass. That's what I mean by "Needs Tests". There's nothing to prevent a later PR from unintentionally breaking this.
I've created https://github.com/raygard/Pillow/pull/1 as my suggestion, with passes on all platforms. You can see at that https://github.com/radarhere/Pillow/actions/runs/562702893 without your change, I can only create a test file that passes on some platforms, but not others.

Reply:
Your suggestion about GifEncode.c has now been merged in as #5265